crates/epaint/src/tessellator.rs

```diff
@@ -864,19 +875,58 @@ fn stroke_path(
     feathering: f32,
     path: &[PathPoint],
     path_type: PathType,
-    stroke: Stroke,
+    stroke: &PathStroke,
     out: &mut Mesh,
 ) {
     let n = path.len() as u32;
 
-    if stroke.width <= 0.0 || stroke.color == Color32::TRANSPARENT || n < 2 {
+    if stroke.width <= 0.0 || stroke.color == ColorMode::Solid(Color32::TRANSPARENT) || n < 2 {
         return;
     }
 
     let idx = out.vertices.len() as u32;
 
+    let min_x = path
+        .iter()
+        .map(|point| OrderedFloat::from(point.pos.x))
+        .min()
+        .map(OrderedFloat::into_inner)
+        .unwrap();
+    let max_x = path
+        .iter()
+        .map(|point| OrderedFloat::from(point.pos.x))
+        .max()
+        .map(OrderedFloat::into_inner)
+        .unwrap();
+    let min_y = path
+        .iter()
+        .map(|point| OrderedFloat::from(point.pos.y))
+        .min()
+        .map(OrderedFloat::into_inner)
+        .unwrap();
+    let max_y = path
+        .iter()
+        .map(|point| OrderedFloat::from(point.pos.y))
+        .max()
+        .map(OrderedFloat::into_inner)
+        .unwrap();
+
+    let get_color = |col: &ColorMode, pos: Pos2| match col {
+        ColorMode::Solid(col) => *col,
+        ColorMode::UV(fun) => {
+            let x = remap_clamp(pos.x, min_x..=max_x, 0.0..=1.0);
+            let y = remap_clamp(pos.y, min_y..=max_y, 0.0..=1.0);
+            fun(pos2(x, y))
+        }
+        ColorMode::UVBounds(rect, fun) => {
+            let x = remap_clamp(pos.x, rect.min.x..=rect.max.x, 0.0..=1.0);
+            let y = remap_clamp(pos.y, rect.min.y..=rect.max.y, 0.0..=1.0);
+            fun(pos2(x, y))
+        }
```

Comment:
I suggest we keep this simple. We can always have constructors to `ColorMode` that does helpful coordinate transforms

```suggestion
        ColorMode::UV(fun) => {
            fun(bbox, pos2(x, y))
        }
```

Reply:
resolved with 66644820a93a76758da3d8f8f5d7b03ba88f79a9